### DISCUSSION TITLE
All batches are finished

### DISCUSSION THREAD

**Discussion by savasgok** [rpldy/react-uploady#543] [Q&A] [answered]
How can I detect all batches are completed?  I need something like when first batch is started I will show alert to user and when the last batch is finalized then will show alert upload completed. Could you kindly help me?

**Comment by yoavniran** [comment, marked as answer]
Hi @savasgok 
You can use the BATCH-START event ([useBatchStartListener](https://react-uploady.org/docs/api/hooks/useBatchStartListener/)) or BATCH-ADD ([useBatchAddListener](https://react-uploady.org/docs/api/hooks/useBatchAddListener/)) and BATCH-FINISH ([useBatchFinishListener](https://react-uploady.org/docs/api/hooks/useBatchFinishListener/)) to keep track of how many batches started and how many finished (in your application's state). When the finished count reaches the started count, you can consider the upload complete at that point (until new uploads are added).
